### DISCUSSION TITLE
Error reports via sending emails to operators

### DISCUSSION THREAD

**Discussion by vlbi-jun-yang** [nvi-inc/fs#181] [Ideas]
`FS` do provide a window to show various Errors. This requires operators to keep en eye on the window regularly. At Onsala, we run VLBI observations without dedicated operators. So, suport scientists have to monitor the status of VLBI observations remotely from time to time. 

Our antenna control software `BIFROST` has a function to send severe errors errors to support scientists via emails.  This is a very helpful function for both developers and support scientists to test and monitor the system more efficently. Thus, I propose to implement a similar function to the future FS.  The function may requires a new control file to define error types/codes and target email addresses.

**Comment by wehimwich** [comment]
This is a good idea. There is an existing hook that could be used for this, the _sterp_ station program. A skeleton for that can be found in _/usr2/st.default/st-0.0.0/sterp/_. The skeleton just provides the structure, the code to report the errors has to be provided by the stations. Some stations already use this to make errors more visible for operators. 

Since sending email is fairly universally available, it may make sense to provide an example _sterp_ that can do that. As you suggest, a control file would be desirable. Specifying the error codes that should have email sent could be similar to what is in _erchk.ctl_. Specifying email addresses would require an expansion. A first cut, which may be sufficient, could be send emails to all the addresses listed. (Note that the errors turned off with `tnx` don't reach _sterp_.)

OTOH, maybe it would make sense to do something more modern and as a separate facility. You could have a program that subscribes to the display stream (which `tnx` affects) or the log stream (which `tnx` does not); using `fsclient -n` for the former and  _streamlog_ (#64) for the latter. Maybe it is better to pipe the output of one of those commands to the new program. The program itself could be written in any convenient language, maybe _python_. That might make it easier to modify and possibly use other more modern messaging systems. This approach would really only be intended for use with the display server.

**Reply by vlbi-jun-yang** [reply]
Thanks for the comments. I did not see the file `erchk.ctl ` in our FS computers. Is there a more general document available for me to learn how FS deal with errrors? 

There might be another way to input the email adress. The `snap` command `op` may be extended further to accept another input: the email address. So, the target operator and email adress may be changed dynamically during the observations.

**Reply by vlbi-jun-yang** [reply]
I have just noticed that the control file `erchk.ctl` is a new feature provided by the FS 10.1.0.beta2. So, we need upgrade our FS to the latest version.
